app/src/main/java/org/mozilla/fenix/home/collections/CollectionViewHolder.kt

```diff
@@ -0,0 +1,194 @@
+            composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
+
+            val menuItems = getMenuItems(
+                collection = validCollection,
+                interactor = interactor
+            )
+
+            Column {
+                Spacer(Modifier.height(12.dp))
+
+                Collection(
+                    collection = validCollection,
+                    expanded = isExpanded,
+                    menuItems = menuItems,
+                    interactor = interactor
+                )
+            }
+        }
+    }
+
+    /**
+     * Dynamically replace the current [TabCollection] shown in this `ViewHolder`.
+     *
+     * @param collection [TabCollection] to be shown
+     * @param expanded Whether to show the collection as expanded or collapsed.
+     */
+    fun bindSession(collection: TabCollection, expanded: Boolean) {
+        this.collection = collection
+        isExpanded = expanded
+    }
+
+    companion object {
+        val LAYOUT_ID = View.generateViewId()
+        const val maxTitleLength = 20
+    }
+}
+
+/**
+ * Rectangular shape with all corners rounded used to display a collapsed collection.
+ */
+private val collapsedCollectionShape = RoundedCornerShape(8.dp)
+
+/**
+ * Rectangular shape with only the top corners rounded used to display an expanded collection with other views
+ * placed immediately below this which can be shown immediately next to it, with no visible separation.
+ */
+private val expandedCollectionShape = RoundedCornerShape(topStart = 8.dp, topEnd = 8.dp)
+
+@Composable
+private fun Collection(
+    collection: TabCollection,
+    expanded: Boolean,
+    menuItems: List<CollectionMenuItemData>,
+    interactor: CollectionInteractor,
+) {
+    var isMenuShown by remember { mutableStateOf(false) }
+    val isExpanded by remember { mutableStateOf(expanded) }
+
+    Card(
+        modifier = Modifier
+            .semantics(mergeDescendants = true) {}
+            .clickable { interactor.onToggleCollectionExpanded(collection, !isExpanded) }
+            .height(48.dp),
+        shape = if (isExpanded) expandedCollectionShape else collapsedCollectionShape,
+        backgroundColor = FirefoxTheme.colors.layer2,
+        elevation = 5.dp, // This needs to match the elevation of the TabInCollectionViewHolder for the shadow.
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxHeight(),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Icon(
+                painter = painterResource(R.drawable.ic_tab_collection),
+                contentDescription = null,
+                modifier = Modifier.padding(
+                    start = 16.dp,
+                    end = 8.dp // (24.dp - 16.dp) hardcoded in ExpandableListHeader
+                ),
+                tint = Paint().apply {
+                    color = Color(collection.getIconColor(LocalContext.current))
```

Comment:
should we maybe get a function in `FirefoxTheme` that handles this? that way we also can just do something like `tint = FirefoxTheme.colors.getCollectionColor()`

Reply:
The color seems to [depend on the particular collection for which it will be used](https://github.com/mozilla-mobile/fenix/blob/f924f744348006e2cb01fb172486fc7bfb1b67dc/app/src/main/java/org/mozilla/fenix/ext/TabCollection.kt#L19).
Case in which the current approach of it being an extension function to the collection seems a bit cleaner by having a smaller scope - if specs change about the collection color we wouldn't need to modify the app theme.

Reply:
gotcha gotcha. part of my concern was this function call not working within a preview, so fingers crossed